controlnext/views.py

```diff
@@ -240,33 +241,40 @@ def rain(self, t0):
         }
 
 
-class ControlnextLoginView(LoginView):
-    default_redirect = "/controlnext/"
-    template_name = 'controlnext/controlnextlogin.html'
+class RedirectAfterLoginView(APIView):
 
-    def post(self, request, *args, **kwargs):
-        form_class = self.get_form_class()
-        form = self.get_form(form_class)
-        if form.is_valid():
-            login(self.request, form.get_user())
-            if request.session.test_cookie_worked():
-                request.session.delete_test_cookie()
-            username = form.cleaned_data['username']
-            user = User.objects.get(username=username)
-            try:
-                grower_url_slug = UserProfile.objects.get(user=user).grower\
-                    .random_url_slug
-                next_url = "/controlnext/" + grower_url_slug
-            except ObjectDoesNotExist:
-                next_url = "/controlnext/login_error"
+    def get(self, request, *args, **kwargs):
+        next_url = request.GET.get('next', False)
+        if next_url:
             return HttpResponseRedirect(next_url)
-        return self.form_invalid(form)
+        try:
+            basins_query = GrowerInfo.objects.filter(
```

Comment:
Is een base class met zo'n `def basin(self)` property geen goed idee? Ik zie het op meer plekken terugkomen.